### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Zilola-Nazarova**
# Hello and welcome to this PR! 👋🏻 
### General requirements:
- there are no linter errors;
- correct flow Gitflow is used;
- work is documented in a professional way.

### Project requirements:
I created a method called current_user in ApplicationController to make current user data available to all controllers:
```
def current_user
    @first_user = User.first
  end
end
```
I created corresponding routes:
```
resources :posts, only: [:index, :new, :create, :show] do
  resources :comments, only: [:new, :create]
  resources :likes, only: [:create]
end
```
I created forms to perform the following functions:
- Create a Post on behalf of the current_user:
  - `app/views/posts/new.html.erb`
  - #new and #create actions in `posts_controller.rb`
- Create a comment on behalf of the current_user:
  - `app/views/comments/new.html.erb`
  - #new and #create actions in `comments_controller.rb`
- Allow Users to add likes to Posts:
  - `app/views/likes/_like_action.html.erb`
  - #create action in `likes_controller.rb`

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -3,10 +3,26 @@ class PostsController < ApplicationController
...
+    @post.author = current_user
+    if @post.save
+      flash[:success] = 'Post created successfully!'
+      redirect_to user_posts_url
+    else
+      flash.now[:error] = 'Error: Post could not be created!'
+      render :new, locals: { post: @post }
+    end
+  end
 end
```
Comment (Munsa1) [inline app/controllers/posts_controller.rb:28]: - [x] (Optional) Consider adding the  strong parameters, here is a link to a resource: [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @comment.user = current_user
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+      redirect_to user_post_path(id: @comment.post_id, user_id: @comment.user_id)
+    else
+      flash.now[:error] = 'Error: Comment could not be created!'
+      render :new, locals: { comment: @comment }
+    end
+  end
+end
```
Comment (Munsa1) [inline app/controllers/comments_controller.rb:18]: - [x] (Optional) Consider adding the  strong parameters, here is a link to a resource: [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).

**Review by Munsa1 — APPROVED** [review]
Hi @Zilola-Nazarova, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Able To Add Likes :heavy_check_mark: 
Able To Post And  Comment :heavy_check_mark: 
Clean CodeBase :heavy_check_mark: 

## Optional suggestions
- [ ] Check inline comments

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Zilola-Nazarova) [reply app/controllers/posts_controller.rb:28]: Thank you @Munsa1 !
Here I am using strong parameters on line 18, but I haven't encapsulated them. I'll do that definitely when I create #update method to keep my code DRY.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Zilola-Nazarova) [reply app/controllers/comments_controller.rb:18]: Thank you @Munsa1 !
Here I am using strong parameters on line 8, but I haven't encapsulated them. I'll do that definitely when I create #update method to keep my code DRY.

**Pull request merged**